apis/pipelines/v1alpha5/runconfiguration_types.go

```diff
@@ -111,24 +111,26 @@ func (rc *RunConfiguration) GetDependencyRuns() map[string]RunReference {
 }
 
 func (rc *RunConfiguration) GetReferencedRCArtifacts() []RunConfigurationRef {
-	return pipelines.Collect(rc.Spec.Run.RuntimeParameters, func(rp RuntimeParameter) (RunConfigurationRef, bool) {
-		if rp.ValueFrom == nil {
-			return RunConfigurationRef{}, false
-		}
-
-		return rp.ValueFrom.RunConfigurationRef, true
-	})
+	return pipelines.Collect(
+		rc.Spec.Run.RuntimeParameters,
+		func(rp RuntimeParameter) (RunConfigurationRef, bool) {
```

Comment:
It was pre-existing, but what is the `bool` return being used for here?

Reply:
Not a great signature. Looks like it's a way of signalling to the `Collect` utility function. Since `ValueFrom` is a ptr, it's trying to see if it can be put into a hashmap. I reckon this can be dealt with if `RunConfigurationRef` is returned as a ptr instead

Reply:
yeah gets used as `canMap` in [here](https://github.com/sky-uk/kfp-operator/blob/5e41f9df0865e4ddede262c3e0f9f81ae1d77ef5/apis/pipelines/utils.go#L101), which I don't think we should change as part of this